provider/cmd/crd2pulumi/README.md

```diff
@@ -1,148 +1,243 @@
+## Usage
 
-```yaml
-apiVersion: apiextensions.k8s.io/v1
-kind: CustomResourceDefinition
-metadata:
-  # name must match the spec fields below, and be in the form: <plural>.<group>
-  name: crontabs.stable.example.com
-spec:
-  # group name to use for REST API: /apis/<group>/<version>
-  group: stable.example.com
-  # list of versions supported by this CustomResourceDefinition
-  versions:
-    - name: v1
-      # Each version can be enabled/disabled by Served flag.
-      served: true
-      # One and only one version must be marked as the storage version.
-      storage: true
-      schema:
-        openAPIV3Schema:
-          type: object
-          properties:
-            spec:
-              type: object
-              properties:
-                cronSpec:
-                  type: string
-                image:
-                  type: string
-                replicas:
-                  type: integer
-  # either Namespaced or Cluster
-  scope: Namespaced
-  names:
-    # plural name to be used in the URL: /apis/<group>/<version>/<plural>
-    plural: crontabs
-    # singular name to be used as an alias on the CLI and for display
-    singular: crontab
-    # kind is normally the CamelCased singular type. Your resource manifests use this.
-    kind: CronTab
-    # shortNames allow shorter string to match your resource on the CLI
-    shortNames:
-    - ct
+```bash
+$ crd2pulumi <language> <crd file path> [output directory] [--force]
 ```
-And ran the command:
+
+`<language>` is the target language to generate code for, so either `nodejs` or `go`.
+
+`<crd file path>` is the path to the k8s CRD YAML file.
+
+`[output directory]` is an optional path to the directory in which to output the
+code to. If this field is not specified, then crd2pulumi will automatically output
+to the same directory as the CRD YAML file.
+
+`[--force]` is an optional flag to overwrite existing files if crd2pulumi would
+write to directories or files that already exist. By default this is set to false.
+
+## Examples
+Let's use the example CronTab CRD specified in `resourcedefinition.yaml` from the [Kubernetes Documentation](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/). 
+
+### Output to NodeJS
+
+To generate a strongly-typed CronTab CustomResource in NodeJS, we can run this command:
 ```bash
-$ go run . resourcedefinition.yaml
+$ crd2pulumi nodejs resourcedefinition.yaml
 ```
-It would generate a file called `crontabs.ts` by default:
+By default, this will create the folder `./crontabs` in the same directory as `resourcedefinition.yaml`. Each versioned CustomResource will live under this folder, so since we had just specified `v1`, you'll see that a `./crontabs/v1` directory has been created. If we had defined a `v2` version in the CRD YAML file, you'd also see `./crontabs/v2`.
```

Comment:
Nit: blank line

```suggestion

By default, this will create the folder `./crontabs` in the same directory as `resourcedefinition.yaml`. Each versioned CustomResource will live under this folder, so since we had just specified `v1`, you'll see that a `./crontabs/v1` directory has been created. If we had defined a `v2` version in the CRD YAML file, you'd also see `./crontabs/v2`.
```